### DISCUSSION TITLE
keras 3.3.3 AttributeError: 'Functional' object has no attribute 'distribute_strategy'

### DISCUSSION THREAD

**Discussion by jeffDebug** [keras-team/keras#19869] [Q&A] [answered]
I ran into this error when I was running the training, with the backend set to torch
` File "C:\jeff\conda\envs\keras3\lib\site-packages\torch\nn\modules\module.py", line 1709, in __getattr__
    raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'Functional' object has no attribute 'distribute_strategy'`

I have installed:
keras==3.3.3
torch==2.3.0+cu118

Here is a small test code that I ran and reproduced this result. If I had misused the keras API, or I missed any configuration for keras 3+, please do let me know. Thanks!

```
def test_model_backend(backend):
    import os
    os.environ["KERAS_BACKEND"] = backend

    from keras.layers import Input, Conv2D
    from keras.models import Model

    inp = Input(shape=(28, 28, 1), batch_size=32)
    oup = Conv2D(filters=10, kernel_size=(3, 3))(inp)
    model = Model(inputs=inp, outputs=oup)

    attr = "distribute_strategy"
    print("Backend:", backend, flush=True)
    print("keras model:", type(model), flush=True)
    print(f"keras model has attr '{attr}':", hasattr(model, attr), flush=True)
    return None
 
if "__main__" == __name__:
    import sys

    backend = sys.argv[1]
    test_model_backend(backend)
```

Output with backend set to torch:

> % python tmp\keras_3\test_1.py torch
> Backend: torch
> keras model: <class 'keras.src.models.functional.Functional'>
> keras model has attr 'distribute_strategy': **_False_**

Output with backend set to tensorflow:

> % python tmp\keras_3\test_1.py tensorflow
> Backend: tensorflow
> keras model: <class 'keras.src.models.functional.Functional'>
> keras model has attr 'distribute_strategy': **_True_**

**Comment by fchollet** [comment, marked as answer]
"distribution strategy" is a TensorFlow concept, so this attribute only exist with the TensorFlow backend. To do distributed training with the torch backend, see this guide: https://keras.io/guides/distributed_training_with_torch/
